java/ql/src/Security/CWE/CWE-319/UseSSLSocketFactories.ql

```diff
@@ -16,23 +16,24 @@ import semmle.code.java.security.Encryption
 
 class NetworkClass extends Class {
   NetworkClass() {
-    this.getAnAncestor().getQualifiedName().matches("java.rmi.%") or
-    this.getAnAncestor().getQualifiedName().matches("java.net.%") or
-    this.getAnAncestor().getQualifiedName().matches("javax.net.%")
+    this.getAnAncestor()
+        .getPackage()
+        .getName()
+        .regexpMatch("(java\\.net|java\\.rmi|javax\\.net)(\\..*)?")
   }
 }
 
 class SocketFactoryType extends RefType {
   SocketFactoryType() {
-    this.getQualifiedName() = "java.rmi.server.RMIServerSocketFactory" or
-    this.getQualifiedName() = "java.rmi.server.RMIClientSocketFactory" or
-    this.getQualifiedName() = "javax.net.SocketFactory" or
-    this.getQualifiedName() = "java.net.SocketImplFactory"
+    this.hasQualifiedName("java.rmi.server", "RMIServerSocketFactory") or
+    this.hasQualifiedName("java.rmi.server", "RMIClientSocketFactory") or
+    this.hasQualifiedName("javax.net", "SocketFactory") or
+    this.hasQualifiedName("java.net", "SocketImplFactory")
   }
 }
 
 /** Holds if the method `m` has a factory parameter at location `p`. */
-cached
+pragma[nomagic]
```

Comment:
Just to be sure: what are the implications of this in terms of performance? Isn't this a bit out of place in an otherwise harmless-looking PR?

Reply:
Firstly, `cached` implies `nomagic`, so when removing a cached annotation, the smallest possible change is to then also add `nomagic`, hence the pragma addition. And caching doesn't really make sense in a `.ql` file - it's not like there's some other query that can reference the predicate, so this was just a drive-by clean-up and should be harmless.

Reply:
Ah, I didn't realize it was a query file. TIL `cached` implies `nomagic` (it's somewhat obvious in hindsight, actually). Thanks for the explanation!